ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -239,6 +240,9 @@ public StoreInfo get(List<? extends StoreKey> ids, EnumSet<StoreGetOptions> stor
     checkStarted();
     // allows concurrent gets
     final Timer.Context context = metrics.getResponse.time();
+    if (ids.size() != new HashSet<>(ids).size()) {
```

Comment:
Do the new HashSet creation only if `ids.size() > 1`?

Reply:
Good suggestion - done